### ISSUE TITLE
Adopt Jetty 12.0.3

### ISSUE THREAD

**Issue by nitind** [eclipse-servertools/servertools#3]
Eclipse Platform after 4.30M1 includes Jetty 12.0.1, so we need to adopt it to keep in sync with them. Typically this starts with @merks graciously packaging them for us.

**Comment by merks** [comment]
It’s on my todo list.  I’m not sure they have a p2 repo for it.  Please 🙏 feel free to ping me on this issue later in the week as a friendly reminder.

**Comment by nitind** [comment]
Updating the version, because the Platform apparently has again.

**Comment by nitind** [comment]
Hmm, part of the porting effort involved me trying to install everything that was in the maven-jetty site provided by @merks (in an attempt to understand the new runtime structure for assembling a viable servlet container), but that actually fails with:
`
Cannot complete the install because one or more required items could not be found.
	Software being installed: EE10 :: Websocket :: Jakarta Common 12.0.3 (org.eclipse.jetty.ee10.websocket.jakarta.common 12.0.3)
	Missing requirement: EE10 :: Websocket :: Jakarta Common 12.0.3 (org.eclipse.jetty.ee10.websocket.jakarta.common 12.0.3) requires 'java.package; jakarta.websocket [2.1.1,3.0.0)' but it could not be found
`

Modern Orbit has jakarta.websocket 2.0.0, with no replacement in simrel-maven as of yet, but then I'm not sure any websocket APIs worked in the preview server before.

**Comment by merks** [comment]
Note that the Platform is using ee8 parts:

https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/b918a4766cd48a4597b52079cfd8caba57c55ac5/eclipse.platform.releng.prereqs.sdk/eclipse-sdk-prereqs.target#L279-L302

I have no idea if that has any impact on what you use.

I don't understand the comment about "no replacement in simrel-maven".  Where is it you are looking and not seeing a replacement?  There are these versions available:

![image](https://github.com/eclipse-servertools/servertools/assets/208716/a20cb76d-e9b6-455e-95a5-22d3b10876de)

One of the tests I do is to ensure that all of Jetty's dependencies are satisfied with this being the complete list of dependencies:

https://github.com/eclipse-orbit/orbit-simrel/blob/3eb7e4979e9675500f158844b92d30b4cea7ab0b/maven-jetty/tp/MavenJetty.target#L799-L1314

So there should be nothing missing.  

Maybe one point of confusion is that the bundle names aren't always obvious from the Maven coordinates.  So for https://repo1.maven.org/maven2/jakarta/websocket/jakarta.websocket-api/1.1.2/ one might expect a bundle named jakarta.websocket-api but if you look in the jar, the name is not what you'd expect: 

![image](https://github.com/eclipse-servertools/servertools/assets/208716/01813472-60db-4e34-b628-1f2d82adae59)

Recall that you can search a repository for the packages it provides:

![image](https://github.com/eclipse-servertools/servertools/assets/208716/0850f70a-c6a2-494e-b8d9-8c07b32fe06a)

**Comment by nitind** [comment]
> Note that the Platform is using ee8 parts:
> ...
> I have no idea if that has any impact on what you use.

Only that that's the version (?) I had to start with. It would be great to leverage their new setup to simultaneously offer Java EE 8 and Jakarta EE 9 and 10 Preview Servers somewhere down the line.

> I don't understand the comment about "no replacement in simrel-maven". Where is it you are looking and not seeing a replacement? There are these versions available:

It's always in the last place I look. I remembered the simrel-maven and maven-jetty sites, but not the orbit-aggregation, the existence of all of which I can't thank you enough for. I've set aside any notions of adding websocket API support for this release, so much of it consumed just keeping up with the Platform changes.
